core/partitioning/partitioning.cpp

```diff
@@ -0,0 +1,250 @@
+
+  std::vector<torch::jit::IValue> jit_inputs_ivalues;
+
+  // set inputs ivalues, now supports Tensor/Int to pass argumentes between different segments
+  for (auto& input : seg_block.raw_inputs()) {
+    TRTORCH_CHECK(ivalues_maps.count(input), "Could not find mini graph input IValue " << input->debugName());
+    if (input->type()->isSubtypeOf(torch::jit::TensorType::get())) {
+      jit_inputs_ivalues.push_back(ivalues_maps[input].toTensor());
+    } else if (input->type()->isSubtypeOf(torch::jit::IntType::get())) {
+      jit_inputs_ivalues.push_back(ivalues_maps[input].toInt());
+    } else if (input->type()->isSubtypeOf(torch::jit::BoolType::get())) {
+      jit_inputs_ivalues.push_back(ivalues_maps[input].toBool());
+    } else if (input->type()->kind() == torch::jit::TypeKind::ListType) {
+      jit_inputs_ivalues.push_back(ivalues_maps[input].toList());
+    } else {
+      TRTORCH_CHECK(input->type()->kind() == torch::jit::TypeKind::TupleType, "Input for mini graph is not TupleType.");
+      jit_inputs_ivalues.push_back(ivalues_maps[input].toTuple());
+    }
+  }
+
+  // run segments to get outputs for later segments input shape, and other arguments such as Int
+  std::vector<torch::jit::IValue> jit_results;
+  torch::jit::IValue jit_results_ivalues = cur_mod.forward(jit_inputs_ivalues);
+  if (jit_results_ivalues.isTuple()) {
+    auto results = jit_results_ivalues.toTuple()->elements();
+    for (auto r : results) {
+      jit_results.push_back(r);
+    }
+  } else {
+    jit_results.push_back(jit_results_ivalues);
+  }
+
+  size_t idx = 0;
+  for (auto& output : seg_block.raw_outputs()) {
+    ivalues_maps[output] = jit_results[idx++];
+  }
+
+  // set input shape for each segmented block so we wil use it in conversion process
+  std::vector<nvinfer1::Dims> input_shape;
+  for (auto& i : seg_block.raw_inputs()) {
+    if (ivalues_maps[i].isTensor()) {
+      input_shape.push_back(util::toDims(ivalues_maps[i].toTensor().sizes()));
+    }
+  }
+
+  seg_block.register_inshape(input_shape);
+}
+
+std::vector<torch::jit::IValue> generateRandomInputs(std::vector<conversion::InputRange>& input_ranges) {
+  // generate random inputs for running pytorch segments
+  std::vector<torch::jit::IValue> random_inputs;
+  for (auto& input_range : input_ranges) {
+    auto cur_shape = input_range.input_shape;
+    std::vector<int64_t> shape;
+    shape.insert(shape.begin(), std::begin(cur_shape.d), std::begin(cur_shape.d) + cur_shape.nbDims);
+    auto in = at::randint(5, shape, {at::kCUDA});
+    random_inputs.push_back(in.clone());
+  }
+  return random_inputs;
+}
+
+void registerSegmentsInputsOutputs(
+    std::vector<SegmentedBlock>& segmented_blocks,
+    std::shared_ptr<torch::jit::Graph> g) {
+  // find the corresponding raw values in original global graph for this segmented block's inputs/outputs
+  std::set<torch::jit::Value*> input_values;
+  for (auto& seg_block : segmented_blocks) {
+    seg_block.registerInputs();

```

Comment:
Can you explain why we have this function `registerInputs` ?
It looks like when we call `SegmentedBlock(kTensorRT, tensorrt_nodes)` , it will create a mini_graph `g_` with all nodes appended. 
Do we actually need a separate registerInputs() to add the graph_inputs to a `inputs_` vector ? 
Is there a problem with doing this in the constructor itself ?

Reply:
That's a good point. I think I used this because I didn't have SegmentedBlock(kTensorRT, tensorrt_nodes) constructer at first. So, I had to construct segmentedBlock first and then append node one by one, after all of this is done we can identify all required inputs and then register inputs. 
Currently if we don't append nodes one by one for SegmentedBlock I think maybe it's better to move this part to our constructor. Thanks!